### PR TITLE
100 edit functionality logic on user profile

### PR DESCRIPTION
**Pull request by Chocolat2**
Add the logic for editing the user profile (NOT the groomer, but can be copy pasted for the most part)

### CODE REVIEW

**Review by chtidaba — DISMISSED** [review]
There is a problem with the feature, when you edit the profile the groomer screen do not fetch correct groomers anymore, and also profile does not update, it's only the database that fetches the good data.
Please try to fix this ASAP

File: app/src/main/java/com/android/PetPamper/CAViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+                          .sortedBy { it.timestamp }
+              inProgressChatMessages.value = false
+            }
+  }
+
+  fun depopulateChat() {
+    chatMessages.value = listOf()
+    currentChatMessagesListener = null
+  }
+}
```
Comment (karinerafla) [inline app/src/main/java/com/android/PetPamper/CAViewModel.kt:200]: chat view model looks good to me.

File: app/src/main/java/com/android/PetPamper/ChannelActivity.kt [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  companion object {
+    private const val KEY_CHANNEL_ID = "channelId"
+
+    fun getIntent(context: Context, channelId: String): Intent {
+      return Intent(context, ChannelActivity::class.java).apply {
+        putExtra(KEY_CHANNEL_ID, channelId)
+      }
+    }
+  }
+}
```
Comment (karinerafla) [inline app/src/main/java/com/android/PetPamper/ChannelActivity.kt:40]: Comments clarify the code.. Thanks

File: app/src/main/java/com/android/PetPamper/MainActivity.kt [hunk 3]
```diff
@@ -40,60 +48,140 @@ import com.android.PetPamper.ui.screen.users.GroomerTopBar
...
-    NavHost(navController = navController, startDestination = "LoginScreen") {
-      composable("LoginScreen") { SignIn(navController) }
+  val members = listOf(userId, groomerId)
+  Log.d("Members", "Members: $members")
+  val extraData = mapOf("name" to "Direct Message between $userId and $groomerId")
 
-      composable("RegisterScreen1") { Register(signUp, navController) }
-      composable("RegisterScreenAlreadyGroomer") { Register(signUp, navController, true) }
+  channelClient.create(members, extraData).enqueue { result ->
+    if (result.isSuccess) {
```
Comment (karinerafla) [inline app/src/main/java/com/android/PetPamper/MainActivity.kt:106]: good separation of tasks. well ordered. comprehensible.

File: app/src/main/java/com/android/PetPamper/Util.kt [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+            fontWeight = FontWeight.Bold,
+            modifier = Modifier.padding(start = 4.dp))
+      }
+}
+
+@Composable
+fun TitleText(txt: String) {
+  Text(
+      text = txt, fontWeight = FontWeight.Bold, fontSize = 35.sp, modifier = Modifier.padding(8.dp))
+}
```
Comment (karinerafla) [inline app/src/main/java/com/android/PetPamper/Util.kt:119]: smart idea to have a Utils class.

File: app/src/main/java/com/android/PetPamper/model/UserViewModel.kt [hunk 5]
```diff
@@ -1,27 +1,99 @@
...
+                }
+            }
+            isLoaded = true
+        }
+    }
+
+    fun updateUser(name:String = user.name, email:String = user.email, phone:String = user.phoneNumber, address:Address = user.address, pawPoints:Int = user.pawPoints, picURL:String = user.profilePictureUrl) {
+        this.user = User(name,email,phone,address,pawPoints,picURL)
+        firebaseConnection.storeData("users",uid,this.user)
+    }
```
Comment (karinerafla) [inline app/src/main/java/com/android/PetPamper/model/UserViewModel.kt:89]: lgtm

File: app/src/main/java/com/android/PetPamper/ui/chat/SingleChatScreen.kt [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+                    unfocusedBorderColor = Color(0xFFBDBDBD)),
+            shape = RoundedCornerShape(16.dp))
+        IconButton(onClick = onSendReply, modifier = Modifier.size(48.dp)) {
+          Icon(
+              imageVector = Icons.Rounded.Send,
+              contentDescription = "Send",
+              tint = Color(0xFF3F51B5))
+        }
+      }
+}
```
Comment (karinerafla) [inline app/src/main/java/com/android/PetPamper/ui/chat/SingleChatScreen.kt:139]: good to have separated chat screen and the list of chats.

File: app/src/main/java/com/android/PetPamper/ui/dialogs/EditProfile.kt [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+                    val phone = _phone.value
+                    onSave(User(name,email,phone,user.address,user.pawPoints,user.profilePictureUrl))
+                }) {
+                Text("Save")
+            }
+        },
+        dismissButton = { androidx.compose.material.Button(onClick = onDismiss) { Text("Cancel") } }).also { Log.d("EditProfile","Alert.also()") }
+
+
+}
```
Comment (karinerafla) [inline app/src/main/java/com/android/PetPamper/ui/dialogs/EditProfile.kt:88]: good implementation. good idea to have a confirm button that saves all changes.

**Review by karinerafla — APPROVED** [review]

**Pull request merged**